scripts/zfs-tests.sh

```diff
@@ -378,6 +378,7 @@ EOF
 	do
 		SINGLETESTDIR=$(dirname "$t")
 		SINGLETESTFILE=$(basename "$t")
+		SINGLETAG=$(echo "$TAGS" | awk -F, '{print $1}')
```

Comment:
The user can still pass in flags and it would throw this off (i.e. a user will call `zfs-tests` with `-T tag1,tag2,...`). The list of tags needs to ultimately be in the format of `tags = [ 'tag1', 'tag2', ... ]`

You could simplify the log by just ignoring the user specified tags entirely and just have `tags = [ ]`. You can then enforce that the user cannot provide both the `-t` and the `-T` option since they are incompatible in that case.

Thoughts?

Reply:
I tested this with multiple, custom tags and it seems to work properly: i think the runfile needs to have at least one tag in common with the one specified from the CLI to be able to run, right? We just take the first one (`-T mytag,2nd,3rd` => `tags = ['mytag']`) when `-t` is specified:

```
root@linux:~# sudo -u nobody -s /usr/share/zfs/zfs-tests.sh -d /var/tmp -v -t tests/functional/atime/atime_001_pos -T mytag,2nd,3rd -I 3

--- Configuration ---
Runfile:         /var/tmp/zfs-tests.1231.run
STF_TOOLS:       /usr/share/zfs/test-runner
STF_SUITE:       /usr/share/zfs/zfs-tests
STF_PATH:        /var/tmp/constrained_path.HtTf
FILEDIR:         /var/tmp
FILES:           /var/tmp/file-vdev0 /var/tmp/file-vdev1 /var/tmp/file-vdev2
LOOPBACKS:       /dev/loop0 /dev/loop1 /dev/loop2 
DISKS:           loop0 loop1 loop2 
NUM_DISKS:       3
FILESIZE:        4G
ITERATIONS:      3
TAGS:            mytag,2nd,3rd
Keep pool(s):    rpool
Missing util(s): fio umask wait 

/usr/share/zfs/test-runner/bin/test-runner.py  -c /var/tmp/zfs-tests.1231.run -T mytag,2nd,3rd -i /usr/share/zfs/zfs-tests -I 3
Test: /usr/share/zfs/zfs-tests/tests/functional/atime/setup (run as root) [00:00] [PASS]
Test: /usr/share/zfs/zfs-tests/tests/functional/atime/atime_001_pos (run as root) [00:10] [PASS]
Test: /usr/share/zfs/zfs-tests/tests/functional/atime/cleanup (run as root) [00:00] [PASS]
Test: /usr/share/zfs/zfs-tests/tests/functional/atime/setup (run as root) [00:00] [PASS]
Test: /usr/share/zfs/zfs-tests/tests/functional/atime/atime_001_pos (run as root) [00:10] [PASS]
Test: /usr/share/zfs/zfs-tests/tests/functional/atime/cleanup (run as root) [00:00] [PASS]
Test: /usr/share/zfs/zfs-tests/tests/functional/atime/setup (run as root) [00:00] [PASS]
Test: /usr/share/zfs/zfs-tests/tests/functional/atime/atime_001_pos (run as root) [00:10] [PASS]
Test: /usr/share/zfs/zfs-tests/tests/functional/atime/cleanup (run as root) [00:00] [PASS]

Results Summary
PASS	   9

Running Time:	00:00:34
Percent passed:	100.0%
Log directory:	/var/tmp/test_results/20171106T181110

root@linux:~# 
```

I agree it doesn't make much sense to accept both `-t` and `-T`, but writing an empty lists of tags (`tags = [ ]`) in the generated runfile will not completely fix this, we also need to change the way `zfs-tests.sh` executes the `test-runner.py` script, otherwise:

```
root@linux:~# cat /var/tmp/zfs-tests.2561.run
[DEFAULT]
pre =
quiet = False
pre_user = root
user = root
timeout = 600
post_user = root
post =
outputdir = /var/tmp/test_results

[tests/functional/atime]
tests = ['atime_001_pos']
pre = setup
post = cleanup
tags = []
root@linux:~# /usr/share/zfs/test-runner/bin/test-runner.py  -c /var/tmp/zfs-tests.2561.run -T mytag,2nd,3rd -i /usr/share/zfs/zfs-tests -I 3
root@linux:~# echo $?
2
root@linux:~# 
```

Reply:
You could set it like this: `tags = ['functional']` if you disallow `-t` and `-T` to be used together. `zfs-tests.sh` uses the default tag of `functional`. So that should work.

Reply:
Good point, i'll try implementing it this way.